mrp_brewing/models/partner.py

```diff
@@ -33,6 +41,27 @@ def compute_last_order_date(self):
                 .mapped('child_ids.sale_order_ids')
                 .filtered(lambda r: r.state not in ['cancel', 'exception']))
             partner_orders = partner_orders + childs_orders
-            partner_orders.sorted()
+            partner_orders = partner_orders.sorted()
+
+            # last orders
             if len(partner_orders) > 0:
                 partner.last_order = partner_orders[0].date_order
+            else:
+                partner.last_order = None
+
+            # sale frequency
+            last_year = dt.datetime.today() - dt.timedelta(days=365)
+            order_dates = partner_orders.mapped('date_order')
+            order_dates = map(fields.Datetime.from_string, order_dates)

```

Comment:
not sure that date_order retrieved from database is in string format. it date_order is in date format you can filter directly on the recordset

Reply:
It is in string format, I was surprised too.